frontend/tasks.py

```diff
@@ -3272,15 +3274,41 @@ def get_rel_energy(s):
 
 
 @app.task(base=AbortableTask)
-def dispatcher(order_id):
-    order = CalculationOrder.objects.get(pk=order_id)
-    ensemble = order.ensemble
+def dispatcher_flowchart():
+    print("Just for testing tasks")
+
+
+@app.task(base=AbortableTask)
+def dispatcher(order_id, drawing=None, is_flowchart=False, flowchartStepObjectId=None):
+    stepFlowchart = None
+    should_create_ensemble = True
+    if is_flowchart:
+        order = FlowchartOrder.objects.get(pk=order_id)
+        flowchartStepObject = Step.objects.get(pk=flowchartStepObjectId)
+        if flowchartStepObject.parentId.step is not None:
+            if flowchartStepObject.parentId.step.creates_ensemble is True:
+                if len(flowchartStepObject.calculation_set.all()) != 0:
```

Comment:
You can use `flowchartStepObject.calculation_set.count()` to get the number of results (it's cleaner and perhaps marginally more efficient).

Reply:
Done!